### PR TITLE
Ci release workflow

### PR DESCRIPTION
**Pull request by tpluscode**
This adds Atlassian changesets and a GitHub workflow to create a release PR + tags when new versions are created. General steps will be

1. Create changeset files using CLI (`yarn changeset`) or directly on GitHub (when we add changeset bot)
2. Merge PR
3. The `release` job creates a PR with new versions in `package.json` files
4. Merge the PR from **3**
5. The `tags` job will create new tags for the CI to pick up and publish docker images

### CODE REVIEW

File: .github/workflows/releases.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      - uses: actions/setup-node@master
+        with:
+          node-version: 14
+
+      - run: yarn
+
+      - name: Create Release Pull Request
+        uses: changesets/action@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (tpluscode) [inline .github/workflows/releases.yaml:27]: The creates and updates a version bump PR

File: .github/workflows/releases.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          root: ui
+          tag_prefix: cube-creator-app/v
+          minVersion: 0.1.14
+
+      - name: Tag CLI
+        uses: tpluscode/action-autotag@min-version-21
+        with:
+          root: cli
+          tag_prefix: cube-creator-cli/v
+          minVersion: 0.1.14
```
Comment (tpluscode) [inline .github/workflows/releases.yaml:55]: This creates tags by extracting the version from `package.json` files and applying the configured prefixes

Using a fork until https://github.com/ButlerLogic/action-autotag/pull/22 would be merged

The latest version is `0.1.13` so the actions are configured to start tagging from `0.1.14`

**Comment by martinmaillard** [comment]
I'm not sure I understand the process 100%. What I have to do to trigger a new release is to run `yarn changeset`? What does  this command do exactly?

**Comment by tpluscode** [comment]
No, `yarn changeset` creates a file where you describe the changes for a given commit/PR and choose affected projects. If you don't commit that file you get an option to create one directly on GitHub too.

All those files combined build up a release

Here's an example output 

```
yarn changeset
yarn run v1.22.10
$ /Users/tomaszpluskiewicz/projects/github/hydrofoil/shaperone/node_modules/.bin/changeset
🦋  Which packages would you like to include? · @hydrofoil/shaperone-wc
🦋  Which packages should have a major bump? · No items were selected
🦋  Which packages should have a minor bump? · @hydrofoil/shaperone-wc
🦋  Please enter a summary for this change (this will be in the changelogs). Submit empty line to open external editor
🦋  Summary · New feature in this package
🦋  === Releasing the following packages ===
🦋  [Minor]
🦋    @hydrofoil/shaperone-wc
🦋  ╔════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════╗
🦋  ║                                                      ========= NOTE ========                                                       ║
🦋  ║All dependents of these packages that will be incompatible with the new version will be patch bumped when this changeset is applied.║
🦋  ╚════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════╝
🦋  Is this your desired changeset? (Y/n) · true
🦋  Changeset added! - you can now commit it
🦋  
🦋  If you want to modify or expand on the changeset summary, you can find it here
🦋  info /Users/tomaszpluskiewicz/projects/github/hydrofoil/shaperone/.changeset/tender-humans-matter.md
✨  Done in 25.10s.
```

**Comment by tpluscode** [comment]
To trigger a release all you will do is merge a PR similar to https://github.com/tpluscode/changest-bot-test/pull/6

**Comment by l00mi** [comment]
So the idea would be to have a (always) PR open for the next release. And add changes there, once the PR gets merged a new release is created?

**Comment by tpluscode** [comment]
Yes, only that the PR is automatically created and updated

**Comment by martinmaillard** [comment]
Ok, I'm fine with trying this approach. I'll take a look at the workflow, but @sandhose is probably better suited to review it.

**Review by martinmaillard — APPROVED** [review]

**Comment by sandhose** [comment]
When tagging, who are you tagging/pushing as? I'm not 100% sure, but I think refs pushed by the GitHub Actions user don't trigger workflows to avoid loops. I had this problem in one of my repo (I was creating a PR with the GH action user, the CI would not trigger on that PR), the workaround was to create a GitHub App and login as the app using [this action](https://github.com/tibdex/github-app-token). Other than that LGTM

**Review by sandhose — APPROVED** [review]

**Comment by tpluscode** [comment]
@sandhose interesting observation. I tried out this configuration in the [linked repository](https://github.com/tpluscode/changest-bot-test) and it was quite happy to run CI on tags created by the actions bot.

**Pull request merged**